// @nuxt/ui/dist/runtime/components/Footer.d.vue.ts
import type { AppConfig } from '@nuxt/schema';
import type { ComponentConfig } from '../types/tv';
import theme from '#build/ui/footer';
type Footer = ComponentConfig<typeof theme, AppConfig, 'footer'>;
export interface FooterProps {
    /**
     * The element or component this component should render as.
     * @defaultValue 'footer'
     */
    as?: any;
    class?: any;
    ui?: Footer['slots'];
}
export interface FooterSlots {
    left(props?: {}): any;
    default(props?: {}): any;
    right(props?: {}): any;
    top(props?: {}): any;
    bottom(props?: {}): any;
}
declare const _default: __VLS_WithSlots<import("vue").DefineComponent<FooterProps, {}, {}, {}, {}, import("vue").ComponentOptionsMixin, import("vue").ComponentOptionsMixin, {}, string, import("vue").PublicProps, Readonly<FooterProps> & Readonly<{}>, {
    as: any;
}, {}, {}, {}, string, import("vue").ComponentProvideOptions, false, {}, any>, FooterSlots>;
export default _default;
type __VLS_WithSlots<T, S> = T & {
    new (): {
        $slots: S;
    };
};
