// @nuxt/ui/dist/runtime/components/Tree.d.vue.ts
import type { TreeRootProps, TreeRootEmits } from 'reka-ui';
import type { AppConfig } from '@nuxt/schema';
import theme from '#build/ui/tree';
import type { IconProps } from '../types';
import type { DynamicSlots, GetItemKeys } from '../types/utils';
import type { ComponentConfig } from '../types/tv';
type Tree = ComponentConfig<typeof theme, AppConfig, 'tree'>;
export type TreeItem = {
    /**
     * @IconifyIcon
     */
    icon?: IconProps['name'];
    label?: string;
    /**
     * @IconifyIcon
     */
    trailingIcon?: IconProps['name'];
    defaultExpanded?: boolean;
    disabled?: boolean;
    slot?: string;
    children?: TreeItem[];
    onToggle?(e: Event): void;
    onSelect?(e?: Event): void;
    class?: any;
    ui?: Pick<Tree['slots'], 'item' | 'itemWithChildren' | 'link' | 'linkLeadingIcon' | 'linkLabel' | 'linkTrailing' | 'linkTrailingIcon' | 'listWithChildren'>;
    [key: string]: any;
};
export interface TreeProps<T extends TreeItem[] = TreeItem[], M extends boolean = false> extends Pick<TreeRootProps<T>, 'expanded' | 'defaultExpanded' | 'selectionBehavior' | 'propagateSelect' | 'disabled' | 'bubbleSelect'> {
    /**
     * The element or component this component should render as.
     * @defaultValue 'ul'
     */
    as?: any;
    /**
     * @defaultValue 'primary'
     */
    color?: Tree['variants']['color'];
    /**
     * @defaultValue 'md'
     */
    size?: Tree['variants']['size'];
    /** This function is passed the index of each item and should return a unique key for that item */
    getKey?: (val: T[number]) => string;
    /**
     * The key used to get the label from the item.
     * @defaultValue 'label'
     */
    labelKey?: GetItemKeys<T>;
    /**
     * The icon displayed on the right side of a parent node.
     * @defaultValue appConfig.ui.icons.chevronDown
     * @IconifyIcon
     */
    trailingIcon?: IconProps['name'];
    /**
     * The icon displayed when a parent node is expanded.
     * @defaultValue appConfig.ui.icons.folderOpen
     * @IconifyIcon
     */
    expandedIcon?: IconProps['name'];
    /**
     * The icon displayed when a parent node is collapsed.
     * @defaultValue appConfig.ui.icons.folder
     * @IconifyIcon
     */
    collapsedIcon?: IconProps['name'];
    items?: T;
    /** The controlled value of the Tree. Can be bind as `v-model`. */
    modelValue?: M extends true ? T[number][] : T[number];
    /** The value of the Tree when initially rendered. Use when you do not need to control the state of the Tree. */
    defaultValue?: M extends true ? T[number][] : T[number];
    /** Whether multiple options can be selected or not. */
    multiple?: M & boolean;
    class?: any;
    ui?: Tree['slots'];
}
export type TreeEmits<T extends TreeItem[] = TreeItem[], M extends boolean = false> = TreeRootEmits<T[number], M>;
type SlotProps<T extends TreeItem> = (props: {
    item: T;
    index: number;
    level: number;
    expanded: boolean;
    selected: boolean;
}) => any;
export type TreeSlots<T extends TreeItem[] = TreeItem[]> = {
    'item-wrapper': SlotProps<T[number]>;
    'item': SlotProps<T[number]>;
    'item-leading': SlotProps<T[number]>;
    'item-label': SlotProps<T[number]>;
    'item-trailing': SlotProps<T[number]>;
} & DynamicSlots<T[number], undefined, {
    index: number;
    level: number;
    expanded: boolean;
    selected: boolean;
}>;
declare const _default: <T extends TreeItem[], M extends boolean = false>(__VLS_props: NonNullable<Awaited<typeof __VLS_setup>>["props"], __VLS_ctx?: __VLS_PrettifyLocal<Pick<NonNullable<Awaited<typeof __VLS_setup>>, "attrs" | "emit" | "slots">>, __VLS_expose?: NonNullable<Awaited<typeof __VLS_setup>>["expose"], __VLS_setup?: Promise<{
    props: __VLS_PrettifyLocal<Pick<Partial<{}> & Omit<{
        readonly "onUpdate:modelValue"?: ((val: M extends true ? T[number][] : T[number]) => any) | undefined;
        readonly "onUpdate:expanded"?: ((val: string[]) => any) | undefined;
    } & import("vue").VNodeProps & import("vue").AllowedComponentProps & import("vue").ComponentCustomProps, never>, "onUpdate:modelValue" | "onUpdate:expanded"> & TreeProps<T, M> & {}> & import("vue").PublicProps;
    expose(exposed: import("vue").ShallowUnwrapRef<{}>): void;
    attrs: any;
    slots: TreeSlots<T>;
    emit: ((evt: "update:modelValue", val: M extends true ? T[number][] : T[number]) => void) & ((evt: "update:expanded", val: string[]) => void);
}>) => import("vue").VNode & {
    __ctx?: Awaited<typeof __VLS_setup>;
};
export default _default;
type __VLS_PrettifyLocal<T> = {
    [K in keyof T as K]: T[K];
} & {};
